tt_metal/impl/device/device.cpp

```diff
@@ -1750,10 +1750,19 @@ LaunchMessageRingBufferState& Device::get_worker_launch_message_buffer_state(Sub
     return this->active_sub_device_manager_->get_worker_launch_message_buffer_state(sub_device_id);
 }
 
+// Main source to get NOC idx for dispatch core
 NOC Device::dispatch_go_signal_noc() const {
     return this->dispatch_s_enabled() ? NOC::NOC_1 : NOC::NOC_0;
 }
 
+NOC Device::prefetcher_noc() const {
+    return NOC::NOC_0;
+}
+
+NOC Device::dispatcher_noc() const {
+    return NOC::NOC_0;
+}
```

Comment:
Do we support these two having different nocs? Otherwise we should assert that they're the same somewhere

Reply:
We don't, good point. I've removed these functions and added a `constexpr` specifying the dispatch NOC in `dispatch_core_common.hpp`. We also now have asserts ensuring that the user specified NOC matches this value for both the prefetcher and dispatcher kernels.